ext/js/application.js

```diff
@@ -190,10 +190,35 @@ export class Application extends EventDispatcher {
      * @param {(application: Application) => (Promise<void>)} mainFunction
      */
     static async main(waitForDom, mainFunction) {
+        /** @type {MessagePort | null} */
+        // If this is Firefox, we don't have a service worker and can't postMessage,
```

Comment:
> "we" here refers to the application a.k.a the iframe within the popup?
Yes.

> How is it that we postMessage to the sharedWorkerBridge but not directly to the backend?
There is simply no way to get access to the backend `Window` object in order to `postMessage` it, which makes sense because direct access to Window would allow for cross-process DOM manipulation which certainly would not work. I looked through all available APIs but could not find any way to do it in a more straight-forward way so I had to use this hack (which I came up with myself btw lol).